c/windows.c

```diff
@@ -268,18 +268,18 @@ static char *s_ErrorStringImp(DWORD dwMessageId, const char *lpcDefault) {
 #undef HEXERRBUFSIZ
         }
     }
-    result = Swide_to_utf8(lpMsgBuf);
-    endstr = result + len;
-    LocalFree(lpMsgBuf);
+    endstr = lpMsgBuf + len;
     /* Otherwise remove trailing newlines & returns and strip trailing period. */
-    while (result != endstr) {
-        char c = *--endstr;
-        if (c == '.') {
+    while (lpMsgBuf != endstr) {
+        wchar_t c = *--endstr;
+        if (c == L'.') {
             *endstr = 0;
             break;
         }
-        else if (c != '\n' && c != '\r') break;
+        else if (c != L'\n' && c != L'\r') break;
```

Comment:
This loop doesn't handle the case where the string doesn't end in a period but has trailing newlines and returns. You might simply use the original loop but with wchar_t to find the new end and then set it to nul.

Reply:
The fact that I wanted to rewrite that part of this code is that it wasn't obvious to me what it was supposed to do. But that wasn't very professional of my part to not test against the old code. I did it this time and I have something that matches 100% what the old code did, except on uninteresting cases (lines solely consisting of newlines and returns).